index.js

```diff
@@ -5,15 +5,47 @@
 
 // This script is modified version of
 // https://github.com/brave/brave-ios/blob/development/Client/Frontend/UserContent/UserScripts/Playlist.js
-(function () {
+medias = (async function () {
   // This will be replaced by native code on demand.
   const siteSpecificDetector = null
 
+  async function isMediaSourceObjectURL (src) {
+    if (!src || !src.startsWith('blob:')) {
+      return false
+    }
+
+    const controller = new AbortController()
+    const signal = controller.signal
+    let timeout
+    const maybeAbortFetch = new Promise(resolve =>
+      timeout = setTimeout(() => resolve(false), 500)
+    )
+
+    return Promise.any([
+      new Promise(resolve => {
+        fetch(src, {
+          signal
+        }).then(response => {
+          resolve(false)
+        }).catch(() => {
+          resolve(true)
+        }).finally(() => {
+          clearTimeout(timeout)
+        })
+      }),
```

Comment:
```suggestion
      fetch(src, { signal })
      .then(() => false)
      .catch(() => true)
      .finally(() => clearTimeout(timeout)),
```

Reply:
I think it's probably fine not to clear the timeout too, as it just resolves the other promise in `Promise.any`, (and aborting a completed promise has no effect)